### PR TITLE
docs: Add catalog-client to project structure

### PR DESCRIPTION
**Pull request by OrkoHunter**
Add new package to https://backstage.io/docs/support/project-structure

### CODE REVIEW

File: docs/support/project-structure.md [hunk 1]
```diff
@@ -89,6 +89,10 @@ are separated out into their own folder, see further down.
   There are no "core" packages in the backend. Instead we have `backend-common`
   which contains helper middleware and other utils.
 
+- [`catalog-client`](https://github.com/backstage/backstage/tree/master/packages/catalog-client) -
+  A client to communicate with Backstage software catalog. Recommended for backend plugins, compatible
```
Comment (benjdlambert) [inline docs/support/project-structure.md:93]: Maybe we can rephrase this to something like:

```
An isomorphic client to interact with the Software Catalog. For use with backend plugins directly, and frontends by using `@backstage/plugin-catalog` in combination with `useApi` and the `catalogApiRef`
```

Comment (OrkoHunter) [reply docs/support/project-structure.md]: @benjdlambert Yeep, very well. Done!

**Review by benjdlambert — APPROVED** [review]
